scripts/o3de/o3de/project_properties.py

```diff
@@ -133,6 +137,8 @@ def add_parser_args(parser):
                        help='Sets the name for the project.')
     group.add_argument('-pid', '--project-id', type=str, required=False,
                        help='Sets the ID for the project.')
+    group.add_argument('-en', '--engine-name', type=str, required=False,
+                       help='Sets the engine name for the project.')
```

Comment:
Once the engine name can be set this pretty much becomes the inverse of `o3de.py register --project-path` command.
Which instead of using the current o3de.py script engine path to lookup the engine name from the engine.json and then register it with the project, this instead allows the user to directly set the engine name.

I am good with this change though, as I thought we needed a command that can update the project's registered engine without needing to go through the register logic.

Reply:
We need a way for a user to register a project with the o3de_manifest and specify the engine it should use without updating the actual `engine.json`and this is one way to do that.  The other option I considered was to modify register.py so you can provide something that indicates you don't want to change `engine.json`  like `o3de.py register --project-path <path> --engine-name <name>`  but this may be confusing to users who may think it's just another way to register it with an engine.